keras/applications/regnet.py

```diff
@@ -0,0 +1,1549 @@
+    bottleneck_ratio: inverted bottleneck ratio 
+    name: name prefix
+
+  Return:
+    Output tensor of the block 
+  """
+  if name is None:
+    name = str(backend.get_uid("zblock"))
+
+  def apply(inputs):
+    if filters_in != filters_out and stride == 1:
+      raise ValueError(
+          f"""Input filters({filters_in}) and output filters({filters_out}) are not equal for stride
+                          {stride}. Input and output filters must be equal for stride={stride}."""
+      )
+
+    groups = filters_out // group_width
+    se_filters = int(filters_in * squeeze_excite_ratio)
+
+    inv_btlneck_filters = int(filters_out / bottleneck_ratio)
+
+    # Build block
+    # conv_1x1_1
+    x = layers.Conv2D(inv_btlneck_filters, (1, 1),
+                      use_bias=False,
+                      kernel_initializer="he_normal",
+                      name=name + "_conv_1x1_1")(inputs)
+    x = layers.BatchNormalization(momentum=0.9,
+                                  epsilon=1e-5,
+                                  name=name + "_conv_1x1_1_bn")(x)
+    x = tf.nn.silu(x)
+
+    # conv_3x3
+    x = layers.Conv2D(inv_btlneck_filters, (3, 3),
+                      use_bias=False,
+                      strides=stride,
+                      groups=groups,
+                      padding="same",
+                      kernel_initializer="he_normal",
+                      name=name + "_conv_3x3")(x)
+    x = layers.BatchNormalization(momentum=0.9,
+                                  epsilon=1e-5,
+                                  name=name + "_conv_3x3_bn")(x)
+    x = tf.nn.silu(x)
+
+    # Squeeze-Excitation block (https://arxiv.org/abs/1709.01507)
+    x = SqueezeAndExciteBlock(inv_btlneck_filters, se_filters, name=name)
+
+    # conv_1x1_2
+    x = layers.Conv2D(filters_out, (1, 1),
+                      use_bias=False,
+                      kernel_initializer="he_normal",
+                      name=name + "_conv_1x1_2")(x)
+    x = layers.BatchNormalization(momentum=0.9,
+                                  epsilon=1e-5,
+                                  name=name + "_conv_1x1_2_bn")(x)
+
+    if stride != 1:
+      return x
+    else:
+      return x + inputs
+
+  return apply
+
+
+def Stage(block_type, depth, group_width, filters_in, filters_out, name=None):
+  """
+  Implementation of Stage in RegNet.
+
+  Args:
+    block_type: must be one of "X", "Y", "Z"
+    depth: depth of stage, number of blocks to use
+    group_width: group width of all blocks in  this stage
+    filters_in: input filters to this stage
+    filters_out: output filters from this stage
+    name: name prefix
+
+  Returns:
+    Output tensor of Stage
+  """
+  if name is None:
+    name = str(backend.get_uid("stage"))
+
+  def apply(inputs):
+    x = inputs
+    if block_type == "X":
+      x = XBlock(filters_in,
+                 filters_out,
+                 group_width,
+                 stride=2,
+                 name=name + "_XBlock_0")(x)
```

Comment:
This code already uses f-strings in a few places, how do you feel about using them for things like this as well?
```suggestion
                 name=f"{name}_XBlock_0")(x)
```